### ISSUE TITLE
Can you add a method to get remote IP for Trackable module?

### ISSUE THREAD

**Issue by hegwin** [plataformatec/devise#4540]
I'm using SLB(Server Load Balancing) serivers that forword requests to my rails servers. In this condition `request.remote_ip` doesn't refers to the real IP of a user but the IP of the SLB machines.

And after I look up into Trackable module, I find I need to define the whole `update_tracked_fields` methods just for a single line.

```ruby
  def update_tracked_fields(request)
    old_current, new_current = self.current_sign_in_at, Time.now.utc
    self.last_sign_in_at     = old_current || new_current
    self.current_sign_in_at  = new_current

    old_current, new_current = self.current_sign_in_ip, (request.headers['X-Forward-For'] || request.remote_ip)
    self.last_sign_in_ip     = old_current || new_current
    self.current_sign_in_ip  = new_current

    self.sign_in_count ||= 0
    self.sign_in_count += 1
  end
```

So can you separate the process to get remote ip to a single method?

Or can I do a fork-and-pull-request? Is that welcome?

**Comment by tegon** [comment]
Hello @hegwin, thanks for your report.
I don't see any problem in doing this. If you're willing to tackle this one, feel free to do it.

Thank you!

**Comment by camilleroux** [comment]
Hi!
Using Cloudflare, I've the same need. Cloudflare use the header `X-Forwarded-For` and can contain IPv6 addresses like:
`X-Forwarded-For: 
2a21:ce1d:824e:900:7529:daaa:b207:a897, 134.156.21.145`

**Comment by mckramer** [comment]
FYI, I tossed together a quick proposal as #4777

**Comment by RouL** [comment]
Just seeing this. Maybe we should only allow this for a list of trusted proxy IPs or when explicitly enabled. If there is no Proxy in front filtering the header, then malicious users could set it for making sure their real ip is not tracked. 

Just a thought.

**Comment by tegon** [comment]
@RouL You mean this should be handled in the application, right? 
When you implement the method to get the IP you can do the security checks that you need.

**Comment by tegon** [comment]
Closed via #4777
